proxy.js

```diff
@@ -171,4 +171,4 @@
 
   scope.Proxy['revocable'] = scope.Proxy.revocable;
   scope['Proxy'] = scope.Proxy;
-})(typeof module !== 'undefined' && module['exports'] ? global : window);
+})(typeof process !== "undefined" && {}.toString.call(process) === "[object process]" ? global : typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope ? self : window);
```

Comment:
I know this is probably correct but its length makes me cry a little inside!

So you want to attach to either: `global`, `self`, or `window`. What about-

```js
typeof module !== 'undefined' && module['exports'] ? global : self
```

as MDN tells me that self is valid for both [Window](https://developer.mozilla.org/en/docs/Web/API/Window/self) and [workers](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/self).